src/GeneralStateTestsFiller/stReturnDataTest/clearReturnBufferFiller.yml

```diff
@@ -0,0 +1,518 @@
+clearReturnBuffer:
+
+  # Make sure that if information is returned by an intermediate step it is not
+  # Returned by subsequent steps.
+
+  env:
+    currentCoinbase: 2adc25665018aa1fe0e6bc666dac8fc2697ff9ba
+    currentDifficulty: 0x20000
+    currentGasLimit: 100000000
+    currentNumber: 1
+    currentTimestamp: 1000
+    previousHash: 5e20a0453cecd065ea59c37ac63e079ee08998b6045136a8ce6635c7912ec0b6
+
+  _info:
+    comment: Ori Pomerantz qbzzt1@gmail.com
+
+  pre:
+
+    # Return a buffer (of a specified length)
+    000000000000000000000000000000000000f3f3:
+      balance: '0x0ba1a9ce0ba1a9ce'
+      code: |
+        :yul {
+           // no need to complicate things with an ABI
+           let bufLen := calldataload(0)
+           mstore(0, 0x60A7)
+           return(0, bufLen)                      
+        }
+      nonce: 1
+      storage: {}
+
+
+
+    # Revert a buffer (of a specified length)
+    000000000000000000000000000000000000fdfd:
+      balance: '0x0ba1a9ce0ba1a9ce'
+      code: |
+        :yul {
+           // no need to complicate things with an ABI
+           let bufLen := calldataload(0)
+           mstore(0, 0x60A7)
+           revert(0, bufLen)       
+        }
+      nonce: 1
+      storage: {}
+
+
+    # This contract is CALLed to receive a buffer
+    # which it then does not return
+    0000000000000000000000000000000000ca11ed:
+      balance: '0x0ba1a9ce0ba1a9ce'
+      code: |
+        :yul {
+           // no need to complicate things with an ABI
+           let addr   := calldataload(0x00)
+           let bufLen := calldataload(0x20)
+           mstore(0, bufLen)
+           pop(call(gas(), addr, 0, 0, 0x20, 0, 0x20))
+           sstore(0, returndatasize())
+           sstore(1, address())
+           stop()
+        }
+      nonce: 1
+      storage: {}
+
+
+    # This contract is CALLed to receive a buffer
+    # which it then does not return. It fails
+    00000000000000000000000000000000bad0ca11:
+      balance: '0x0ba1a9ce0ba1a9ce'
+      code: |
+        :yul {
+           // no need to complicate things with an ABI
+           let addr   := calldataload(0x00)
+           let bufLen := calldataload(0x20)
+           mstore(0, bufLen)
+           pop(call(gas(), addr, 0, 0, 0x20, 0, 0x20))
+
+           // Crash with an illegal opcode
+           verbatim_0i_0o("0xFE")
```

Comment:
is this a function?

Reply:
does it requires a specific solidity version?

Reply:
0.8.5 or above: https://twitter.com/solidity_lang/status/1403033155603476483